crypto/ctype.c

```diff
@@ -278,3 +283,24 @@ int ossl_ascii_isdigit(const char inchar) {
         return 1;
     return 0;
 }
+
+static locale_t loc;
+
+int ossl_init_casecmp_int() {
+    loc = newlocale(LC_COLLATE_MASK, "C", (locale_t) 0);
+    return (loc == (locale_t) 0) ? 0 : 1;
+}
+
+void ossl_deinit_casecmp() {
+    freelocale(loc);
+}
+
+int ossl_strcasecmp(const char *s1, const char *s2)
+{
+    return strcasecmp_l(s1, s2, loc);
+}
+
+int ossl_strncasecmp(const char *s1, const char *s2, size_t n)
+{
+    return strncasecmp_l(s1, s2, n, loc);
+}
```

Comment:
All of the functions used here appeared with POSIX.1-2008...  some platforms we currently build on don't have that.  Is it possible to have a fallback for those?

Reply:
Sure. I will add some fallbacks and native Windows functions like it is done now but 1st I need to solve the FIPS problem, so any help here is highly appreciated.